File: src/main/java/ru/testproject/catalog/model/Product.java
package ru.testproject.catalog.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.sun.istack.NotNull;

import javax.persistence.*;

@Entity
@Table(name = "product")
public class Product extends BasicEntity {
    @Column(nullable = false, unique = true)
    private String name;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "category_id", referencedColumnName = "id")
    @JsonBackReference
    private Category category;

    public Product() {
        super();
    }

    public Product(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }
}

File: src/main/kotlin/ru/testproject/catalog/model/Product.kt
package ru.testproject.catalog.model

import ru.testproject.catalog.model.BasicEntity
import com.fasterxml.jackson.annotation.JsonBackReference
import com.sun.istack.NotNull
import javax.persistence.*

@Entity
@Table(name = "product")
class Product(
    @Column(nullable = false, unique = true)
    var name: String
) : BasicEntity() {

    @NotNull
    @ManyToOne
    @JoinColumn(name = "category_id", referencedColumnName = "id")
    @JsonBackReference
    var category: Category? = null

}
